### PR TITLE
feature(files): adds ElggFile::getPathFromDataRoot

### LINKED ISSUE #9137: Add ElggFile::getPathFromRoot()
We need a new method to quickly get a relative path from data root, otherwise we have to parse it from  `ElggFile::getFilenameOnFilestore()`

### PR DESCRIPTION
**Pull request by hypeJunction**
Adds a convenience method to retrieve relative path to the file from the
filestore root.

Fixes #9137

### CODE REVIEW

**Comment by hypeJunction** [comment]
What happened to all the tests?

File: engine/classes/ElggFile.php [hunk 1]
```diff
@@ -80,6 +80,18 @@ public function getFilenameOnFilestore() {
 	}
 
 	/**
+	 * Returns path relative to the filestore (data directory) root
+	 *
```
Comment (mrclay) [inline engine/classes/ElggFile.php]: Custom filestores means this doesn't guarantee the path will be relative to `dataroot`. Maybe this API isn't such a good idea.

Comment (mrclay) [inline engine/classes/ElggFile.php]: Instead the FileService\File class can just manually verify that filenameOnFilestore() gives back a path containing dataroot and give back a 404 if not...

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: Is there any way to implement a custom filestore at this point? Will it ever be an option given a few thousand plugins are built around filestore being the data root?

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: > Instead the FileService\File class can just manually verify that filenameOnFilestore() gives back a path containing dataroot and give back a 404 if not...

It does already.

File: engine/tests/ElggCoreFilestoreTest.php [hunk 2]
```diff
@@ -48,6 +48,25 @@ public function testFilenameOnFilestore() {
...
+
+		// setup a test file
+		$file = new \ElggFile();
+		$file->owner_guid = $user->guid;
+		$file->setFilename('testing/filestore.txt');
+		$file->open('write');
+		$file->write('Testing!');
+		$this->assertTrue($file->close());
+
+		$this->assertIdentical($file->getPathFromDataRoot(), $dir . 'testing/filestore.txt');
```
Comment (mrclay) [inline engine/tests/ElggCoreFilestoreTest.php]: This test is incomplete, we also would need to test `is_file(elgg_get_data_root() . $file->getPathFromDataRoot())`

File: engine/classes/ElggFile.php (hunk 1 above)
Comment (hypeJunction) [inline engine/classes/ElggFile.php]: Perhaps renaming to `getPathFromFilestoreRoot` and adding `getPathToFilestore` would make it more usable. Plugins can then compare filestore path to dataroot path.

Comment (mrclay) [inline engine/classes/ElggFile.php]: AFAICT devs have been able to `$file->setFilestore($custom_fs)` since 1.0.

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: Oh, wow, what a can of worms... :fishing_pole_and_fish:

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: Let's solicit some feedback from the community. I am not sure how usable that API is in practice. Setting a custom filestore for individual files doesn't seem like something that would work in practice, unless you hack `ege*` with custom callback function that would inject custom filestore.

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: https://elgg.org/discussion/view/2402711/custom-filestore

**Pull request closed without merging**